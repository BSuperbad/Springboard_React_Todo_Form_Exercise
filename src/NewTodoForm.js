import React, {useState} from "react";

const NewTodoForm = ({addNewTodo}) => {
    const INITIAL_STATE = {
        todo_text: "",
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(data => ( {
            ...data,
            [name]: value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addNewTodo({...formData});
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="todo_text">Add New Todo: </label>
            <input 
            id="todo_text"
            type="text" 
            name="todo_text"
            placeholder="Enter a Task"
            value={formData.todo_text}
            onChange={handleChange}
            />
            <button type="submit">Add Task to Todo List</button>
        </form>
    )
}

export default NewTodoForm;