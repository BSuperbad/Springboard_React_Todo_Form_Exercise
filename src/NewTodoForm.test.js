import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders New Todo Form without crashing", function () {
    render(<NewTodoForm />);
});

it("matches New Todo Form snapshot", function() {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

  it("should call addNewTodo function on form submission", () => {
    const addTodoMock = jest.fn();
    const { getByLabelText, getByText } = render(<NewTodoForm addNewTodo={addTodoMock} />);

    fireEvent.change(getByLabelText("Add New Todo:"), { target: { value: "Make Bed" } });

    fireEvent.click(getByText("Add Task to Todo List"));

    expect(addTodoMock).toHaveBeenCalledWith({ todo_text: "Make Bed" });
  });



