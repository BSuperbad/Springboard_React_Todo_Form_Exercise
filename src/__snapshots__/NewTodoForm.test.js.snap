// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches New Todo Form snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo_text"
    >
      Add New Todo: 
    </label>
    <input
      id="todo_text"
      name="todo_text"
      placeholder="Enter a Task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Task to Todo List
    </button>
  </form>
</DocumentFragment>
`;
